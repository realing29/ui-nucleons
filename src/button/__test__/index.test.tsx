import React from 'react';
import { Button, computeClassName } from '../index';
import { shallow } from 'enzyme';
import SearchSVG from '@dev-dep/ui-quarks/icons/24x24/Stroked/search.js';
import PlusSVG from '@dev-dep/ui-quarks/icons/16x16/Stroked/plus.js';

describe('<Button />', () => {
  it('should renders without props', () => {
    const wrapper = shallow(<Button>Кнопка</Button>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render secondary button', () => {
    const wrapper = shallow(<Button actionType='secondary'>Отмена</Button>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render icon button', () => {
    const wrapper = shallow(<Button icon={PlusSVG} />);

    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ iconPosition: 'end' });
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle "size" prop', () => {
    const wrapper = shallow(<Button size='small'>Отмена</Button>);

    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ size: 'medium' });
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle "icon" and "iconPosition" props', () => {
    const wrapper = shallow(<Button icon={SearchSVG}>Икать</Button>);

    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ iconPosition: 'end' });
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ iconPosition: 'start' });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('computeClassName()', () => {
  it('should handle all props', () => {
    expect(computeClassName({
      size: 'small',
      actionType: 'secondary',
      iconPosition: 'end',
      withIcon: true,
      withContent: true,
      className: 'custom-button',
    })).toBe('custom-button button-base button-secondary button-small button-small-with-end-icon');
  });
});
