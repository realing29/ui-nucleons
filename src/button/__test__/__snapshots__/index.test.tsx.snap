// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should handle "icon" and "iconPosition" props 1`] = `
<button
  className="button-base button-primary button-medium button-medium-with-start-icon"
  data-testid="button"
>
  <SearchSVG
    className="icon icon-start"
    height={24}
    width={24}
  />
  Икать
</button>
`;

exports[`<Button /> should handle "icon" and "iconPosition" props 2`] = `
<button
  className="button-base button-primary button-medium button-medium-with-end-icon"
  data-testid="button"
>
  Икать
  <SearchSVG
    className="icon icon-end"
    height={24}
    width={24}
  />
</button>
`;

exports[`<Button /> should handle "icon" and "iconPosition" props 3`] = `
<button
  className="button-base button-primary button-medium button-medium-with-start-icon"
  data-testid="button"
>
  <SearchSVG
    className="icon icon-start"
    height={24}
    width={24}
  />
  Икать
</button>
`;

exports[`<Button /> should handle "size" prop 1`] = `
<button
  className="button-base button-primary button-small text-button-small"
  data-testid="button"
>
  Отмена
</button>
`;

exports[`<Button /> should handle "size" prop 2`] = `
<button
  className="button-base button-primary button-medium text-button-medium"
  data-testid="button"
>
  Отмена
</button>
`;

exports[`<Button /> should render icon button 1`] = `
<button
  className="button-base button-primary button-medium"
  data-testid="button"
>
  <PlusSVG
    className="icon"
    height={24}
    width={24}
  />
</button>
`;

exports[`<Button /> should render icon button 2`] = `
<button
  className="button-base button-primary button-medium"
  data-testid="button"
>
  <PlusSVG
    className="icon"
    height={24}
    width={24}
  />
</button>
`;

exports[`<Button /> should render secondary button 1`] = `
<button
  className="button-base button-secondary button-medium text-button-medium"
  data-testid="button"
>
  Отмена
</button>
`;

exports[`<Button /> should renders without props 1`] = `
<button
  className="button-base button-primary button-medium text-button-medium"
  data-testid="button"
>
  Кнопка
</button>
`;
